Sources/BraveWallet/Panels/Signature Request/SignatureRequestView.swift

```diff
@@ -37,6 +37,67 @@ struct SignatureRequestView: View {
     keyringStore.allAccounts.first(where: { $0.address == currentRequest.address }) ?? keyringStore.selectedAccount
   }
   
+  /// Request display text, used as fallback.
+  private var requestDisplayText: String {
+    if currentRequest.domain.isEmpty {
+      return requestMessage
+    }
+    return """
+    \(Strings.Wallet.signatureRequestDomainTitle):
+    \(requestDomain)
+    
+    \(Strings.Wallet.signatureRequestMessageTitle):
+    \(requestMessage)
+    """
+  }
+  
+  /// Formatted request display text. Will display with bold `Domain` / `Message` headers if domain is non-empty.
+  private var requestDisplayAttributedText: NSAttributedString? {
+    let metrics = UIFontMetrics(forTextStyle: .body)
+    let desc = UIFontDescriptor.preferredFontDescriptor(withTextStyle: .body)
+    let regularFont = metrics.scaledFont(for: UIFont.systemFont(ofSize: desc.pointSize, weight: .regular))
+    if currentRequest.domain.isEmpty {
+      // if we don't show domain, we don't need the full title
+      return NSAttributedString(string: requestMessage, attributes: [.font: regularFont])
+    }
+    let boldFont = metrics.scaledFont(for: UIFont.systemFont(ofSize: desc.pointSize, weight: .bold))
+    
+    let domainTitle = NSAttributedString(string: "\(Strings.Wallet.signatureRequestDomainTitle):\n", attributes: [.font: boldFont])
+    let domain = NSAttributedString(string: requestDomain, attributes: [.font: regularFont])
+    let messageTitle = NSAttributedString(string: "\n\(Strings.Wallet.signatureRequestMessageTitle):\n", attributes: [.font: boldFont])
+    let message = NSAttributedString(string: requestMessage, attributes: [.font: regularFont])
+    
+    let attrString = NSMutableAttributedString(attributedString: domainTitle)
+    attrString.append(domain)
+    attrString.append(messageTitle)
+    attrString.append(message)
+    return attrString
+  }
+  
+  private var requestDomain: String {
+    if showOrignalMessage[requestIndex] == true {
+      return currentRequest.domain
+    } else {
+      let uuid = UUID()
+      var result = currentRequest.domain
+      if needPilcrowFormatted[requestIndex] == true {
+        var copy = currentRequest.domain
+        while copy.range(of: "\\n{2,}", options: .regularExpression) != nil {
```

Comment:
@StephenHeaps Is it possible to add a test for `requestDomain` function and comment about the details of the operatiuon here

Reply:
Discussed over Slack, but some complications when testing this due to the unicode logic with pilcrow/unknown characters and Xcode tests 😔. 
ex.
<img width="660" alt="Screenshot 2023-02-09 at 11 54 16 AM" src="https://user-images.githubusercontent.com/5314553/217889312-03d23883-f098-4ac0-a4db-cf3f23eb82e2.png">
<img width="647" alt="Screenshot 2023-02-09 at 12 20 58 PM" src="https://user-images.githubusercontent.com/5314553/217889675-68bebdb4-7321-4cae-b1bd-3efc2763c4b1.png">